File: app/src/main/java/vip/mystery0/simple_tools/MainActivity.java
package vip.mystery0.simple_tools;

import android.util.Log;

import java.io.File;

import vip.mystery0.tools.base.BaseActivity;
import vip.mystery0.tools.utils.ActivityManagerTools;
import vip.mystery0.tools.utils.ArchiveTools;
import vip.mystery0.tools.utils.CommandTools;
import vip.mystery0.view.ProgressDialog;

public class MainActivity extends BaseActivity {
	private static final String TAG = "MainActivity";

	public MainActivity() {
		super(R.layout.activity_main);
	}

	@Override
	public void bindView() {
		super.bindView();
	}

	@Override
	public void initData() {
		super.initData();
//		final ProgressDialog progressDialog = new ProgressDialog(this);
//		progressDialog.setTitle("title");
//		progressDialog.setMessage("message");
//		progressDialog.show();

//		new Thread(new Runnable() {
//			@Override
//			public void run() {
//				ArchiveTools.Companion.tarGz()
//						.setCompressDir(new File("/sdcard/JanYo Share"))
//						.setArchiveFileName("test")
//						.setSavePath(new File("/sdcard/"))
//						.setSuffix("tar.gz")
//						.compress();

//				runOnUiThread(new Runnable() {
//					@Override
//					public void run() {
//						progressDialog.cancel();
//					}
//				});
//			}
//		}).start();
	}

	@Override
	public void initView() {
		super.initView();
	}

	@Override
	public void requestData() {
		super.requestData();
	}
}

File: app/src/main/java/vip/mystery0/simple_tools/MainActivity.kt
package vip.mystery0.simple_tools

import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*
import vip.mystery0.tools.base.BaseActivity
import vip.mystery0.tools.factory.getExtensionsFromMimeType
import vip.mystery0.tools.factory.toJson
import vip.mystery0.tools.initTools
import vip.mystery0.tools.utils.fastClick

class MainActivity : BaseActivity(R.layout.activity_main) {
	private val TAG = "MainActivity"

	override fun requestData() {
		super.requestData()
		initTools()
		Log.i(TAG, "text/plain".getExtensionsFromMimeType().toJson())
		text_view.setOnClickListener {
			fastClick {
				toast("fastClick")
			}
		}
	}
}
